src/qt/sendcoinsdialog.cpp

```diff
@@ -641,14 +641,18 @@ void SendCoinsDialog::setBalance(const interfaces::WalletBalances& balances)
 {
     if(model && model->getOptionsModel())
     {
-        CAmount bal = 0;
+        CAmount balance = 0;
         if (m_coin_control->IsUsingCoinJoin()) {
-            bal = balances.anonymized_balance;
+            balance = balances.anonymized_balance;
         } else {
-            bal = balances.balance;
+            balance = balances.balance;
         }
 
-        ui->labelBalance->setText(BitcoinUnits::formatWithUnit(model->getOptionsModel()->getDisplayUnit(), bal));
+        if (model->privateKeysDisabled()) {
+            balance = balances.watch_only_balance;
+            ui->labelBalanceName->setText(tr("Watch-only balance:"));
+        }
```

Comment:
17587: this should be smth like
```
        CAmount balance = 0;
        if (model->privateKeysDisabled()) {
            balance = balances.watch_only_balance;
            ui->labelBalanceName->setText(tr("Watch-only balance:"));
        } else if (m_coin_control->IsUsingCoinJoin()) {
            balance = balances.anonymized_balance;
        } else {
            balance = balances.balance;
        }
```
or it's going to show watch-only balance on CoinJoin tab otherwise _I think_